samza-core/src/main/java/org/apache/samza/clustermanager/container/placement/ContainerPlacementMessageObjectMapper.java

```diff
@@ -0,0 +1,119 @@
+    objectMapper.registerSubtypes(new NamedType(ContainerPlacementResponseMessage.class));
+    objectMapper.registerSubtypes(new NamedType(ContainerPlacementRequestMessage.class));
+    return objectMapper;
+  }
+
+  static class ContainerPlacementMessageSerializer extends JsonSerializer<ContainerPlacementMessage> {
+    @Override
+    public void serialize(ContainerPlacementMessage value, JsonGenerator jsonGenerator, SerializerProvider provider)
+        throws IOException {
+      Map<String, Object> containerPlacementMessageMap = new HashMap<String, Object>();
+      if (value instanceof ContainerPlacementRequestMessage) {
+        containerPlacementMessageMap.put("subType", ContainerPlacementRequestMessage.class.getSimpleName());
+      } else if (value instanceof ContainerPlacementResponseMessage) {
+        containerPlacementMessageMap.put("subType", ContainerPlacementResponseMessage.class.getSimpleName());
+        containerPlacementMessageMap.put("responseMessage",
+            ((ContainerPlacementResponseMessage) value).getResponseMessage());
+      }
+      if (value.getRequestExpiry() != null) {
+        containerPlacementMessageMap.put("requestExpiry", value.getRequestExpiry().toMillis());
+      }
+      containerPlacementMessageMap.put("uuid", value.getUuid().toString());
+      containerPlacementMessageMap.put("deploymentId", value.getDeploymentId());
+      containerPlacementMessageMap.put("processorId", value.getProcessorId());
+      containerPlacementMessageMap.put("destinationHost", value.getDestinationHost());
+      containerPlacementMessageMap.put("statusCode", value.getStatusCode().name());
+      containerPlacementMessageMap.put("timestamp", value.getTimestamp());
+      jsonGenerator.writeObject(containerPlacementMessageMap);
+    }
+  }
+
+  static class ContainerPlacementMessageDeserializer extends JsonDeserializer<ContainerPlacementMessage> {
+    @Override
+    public ContainerPlacementMessage deserialize(JsonParser jsonParser, DeserializationContext context)
+        throws IOException {
+      ObjectCodec oc = jsonParser.getCodec();
+      JsonNode node = oc.readTree(jsonParser);
+      String subType = node.get("subType").getTextValue();
+      String deploymentId = node.get("deploymentId").getTextValue();
+      String processorId = node.get("processorId").getTextValue();
+      String destinationHost = node.get("destinationHost").getTextValue();
+      long timestamp = node.get("timestamp").getLongValue();
+      Duration requestExpiry =
+          node.get("requestExpiry") == null ? null : Duration.ofMillis(node.get("requestExpiry").getLongValue());
+      ContainerPlacementMessage.StatusCode statusCode = null;
+      UUID uuid = UUID.fromString(node.get("uuid").getTextValue());
+      for (ContainerPlacementMessage.StatusCode code : ContainerPlacementMessage.StatusCode.values()) {
+        if (code.name().equals(node.get("statusCode").getTextValue())) {
+          statusCode = code;
+        }
+      }
```

Comment:
It is not easy to read/understand, can we do:

statusCode = ContainerPlacementMessage.StatusCode.valueOf(node.get("statusCode").getTextValue())